transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -633,8 +651,13 @@ public final ChannelHandlerContext context(String name) {
         if (name == null) {
             throw new NullPointerException("name");
         }
-
-        return context0(name);
+        synchronized (handlers) {
+            int idx = findCtxIdx((ctx) -> ctx.name().equals(name));
+            if (idx == -1) {
+                return null;
+            }
+            return handlers.get(idx);
```

Comment:
simplify to `return idx == -1 ? null  : handlers.get(idx)`? (same for other similar places)